exercises/joseph_ramsay/ex02_oo_basics/src/shape.h

```diff
@@ -0,0 +1,18 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+
+#include <string>
+
+enum ShapeType { T_square, T_circle, T_rectangle, T_triangle };
```

Comment:
Have a look at strongly typed enum classes. We tend to use them  (for type safety reasons mainly). See [here](http://www.cprogramming.com/c++11/c++11-nullptr-strongly-typed-enum-class.html) for example